From 8f1a08991789cde7652ff4ccb9a9a46abe18d74e Mon Sep 17 00:00:00 2001
From: Leandro Ribeiro <leandro.ribeiro@collabora.com>
Date: Mon, 23 Sep 2024 17:06:31 -0300
Subject: [PATCH 1/4] drm: try other planes that may support fences

Do not skip all the planes if a single one of them do not support
fences. The other may do.

Signed-off-by: Leandro Ribeiro <leandro.ribeiro@collabora.com>
(cherry picked from commit b347af2ce0fca3c7616d5a4d6500ca8770ad3252)
---
 libweston/backend-drm/state-propose.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/libweston/backend-drm/state-propose.c
+++ b/libweston/backend-drm/state-propose.c
@@ -595,7 +595,7 @@ drm_output_find_plane_for_view(struct dr
 		    plane->props[WDRM_PLANE_IN_FENCE_FD].prop_id == 0) {
 			drm_debug(b, "\t\t\t\t[%s] not placing view %p on %s: "
 			          "no in-fence support\n", p_name, ev, p_name);
-			return NULL;
+			continue;
 		}
 
 		if (!b->has_underlay && mm_has_underlay) {
